src/05-derivatives/05-diffusion-derivatives.md

```diff
@@ -18,132 +18,359 @@ should be included in the pipeline documentation.
-| `FORECAST`  | Fiber ORientation Estimated using Continuous Axially Symmetric Tensors. (Zuchelli et al. 2017) | 4D image with SH coefficients.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                |
-| `fwDTI`     | Free water DTI (Hoy et al. 2014)                                                               | One 4D image with parameter name "`tensor`", containing 6 volumes in the order Dc<sub>xx</sub>, Dc<sub>xy</sub>, Dc<sub>xz</sub>, Dc<sub>yy</sub>, Dc<sub>yz</sub>, Dc<sub>zz</sub>, where Dc is the fw-corrected diffusion tensor in each voxel; one 3D image with parameter name "`FWF`" corresponding to the free water fraction in each voxel. .                                                                                                                                                                                                                                                                                                                                                                          |
-| `BedpostX`  | Ball-and-Stick model (Behrens et al. 2007; Jabdbi et al, MRM 2012)                             | 5D image (xyz, m, n) for m parameters and n MCMC samples (n=1 implies best-fit parameters). Parameters are f<sub>i</sub>, th<sub>i</sub>, ph<sub>i</sub> (volume fraction, polar angle, azimuthal angle) for up to 3 fibres, D, D<sub>std</sub> (for “multiexponential” model)                                                                                                                                                                                                                                                                                                                                                                                                                                                |
+-   OPTIONAL file "`<source_keywords>[_space-<space>][_desc-<label>]_parameter-<extparam*>_<model>.json`"
+    provides only information or parameters relevant to derivation of the
+    relevant [extrinsic model parameter](#extrinsic-model-parameters) based
+    on prior estimates of [intrinsic model parameters](#intrinsic-model-parameters).

```

Comment:
Fundamentally the question becomes:

1. How far do two different pieces of data need to differ in their position in the hierarchy of image processing / derivation before they need to be presented explicitly as existing at different positions within that hierarchy?

2. In what way should this variation in position within the hierarchy be encoded?

Clearly we have a distinction between BIDS raw data and BIDS derivatives, which is defined by having a sub-folder for the processing pipeline that produced the derivative. Does the "progression" from intrinsic model parameters to extrinsic model parameters also warrant a bifurcation in order to codify that dependence; and if so, how "large" should the distinction between them be?

I struggled a lot with this when coming up with the proposal, and kinda wish I'd written down my thoughts at the time. But I do think that the fact that some models can blur the lines slightly between intrinsic and extrinsic means that the distinction between the two should be on the smaller side.

As an example: Consider the fibre orientations output by `bedpostx`. The most "native" form of its output (talking specifically about the mean across bootstraps here) is the two spherical angles and fibre volume fraction per stick per voxel. It also outputs the "dyads" images, which are 3-vectors that also encode the fibre orientations. Now the latter are exported by `bedpostx`, so could be considered intrinsic parameters. But if these data were *not* exported by `bedpostx`, it would not be difficult to generate these data from the spherical angle images. Is this therefore then extrinsic parameter data?

An alternative way of looking at it: If examining the contents of a BIDS derivatives directory, it may not actually be that consequential to be able to easily identify which images are intrinsic and which are extrinsic. If the distinction is important in a particular case, then it's likely that JSON files would need to be provided for both, at which point the distinction can be inferred. But if just looking at the data, which are intrinsic and which are extrinsic don't actually matter that much. If looking to detect & automate the further processing of such, that will require knowledge of the spec, at which point those parameters considered intrinsic are known from the spec.

So: While technically yes there is a "derivative" dependence here, I'm not sure that codifying such would actually be beneficial pragmatically, and may in fact come with detriments. Happy for possible alternative solutions to be floated though.